pkgs/tools/misc/hdf4/default.nix

```diff
@@ -31,11 +31,12 @@ stdenv.mkDerivation rec {
     "-DHDF4_BUILD_WITH_INSTALL_NAME=OFF"
     "-DHDF4_ENABLE_JPEG_LIB_SUPPORT=ON"
     "-DHDF4_ENABLE_NETCDF=OFF"
-    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
-    "-DHDF4_ENABLE_SZIP_SUPPORT=ON"
     "-DHDF4_ENABLE_Z_LIB_SUPPORT=ON"
     "-DHDF4_BUILD_FORTRAN=OFF"
     "-DJPEG_DIR=${libjpeg}"
+  ] ++ stdenv.lib.optionals (szip != null) [
+    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
```

Comment:
>They are just bad at writing down what they want.

But they write an explicit prohibition on uses of the code other than encoding and decoding SZIP format. I think what they are writing is close to what they mean.

> In its current form using szip as packaged in NixPkgs is only possible for commercial users with a license.

Of for non-commercial users.

> if someone would e-mail them, they would likely welcome people to distribute binary versions of it

That is probably true, but the NixPkgs policy decision is that we want it to be easy to check if your system is free-software-only (and find exceptions, if any). So even with an explicit permission to distribute binaries, the user would still need to jump through some hoops to install (because if avoiding non-libre software is maximally easy, as it has deliberately been made in NixPkgs, installing non-libre software has to include some explicit override somewhere in the process).

Fixing the free-software-only build of PostGIS is a reasonable goal, and it has been achieved by this PR.